### PR TITLE
feat: 'kpm push' supports the default oci url.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix issue #111 

#### 2. What is the scope of this PR (e.g. component or file name):

pkg/cmd/cmd_push.go

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/cmd/cmd_push.go [hunk 1]
```diff
@@ -67,42 +54,41 @@ func NewPushCmd(settings *settings.Settings) *cli.Command {
 	}
 }
 
-// pushCurrentPackage will push the current package to the oci registry.
-func pushCurrentPackage(ociUrl string, settings *settings.Settings) (string, error) {
-	pwd, err := os.Getwd()
+const OCI_SCHEME = "oci"
```
Comment (Peefy) [inline pkg/cmd/cmd_push.go:57]: Move it to the `oci` paackage.

**Comment by Peefy** [comment]
The kpm push and pull commands should not be bound to OCI, as the default is that the OCI source is OK. Extensibility needs to be reserved through interfaces, such as referring to `helm install/push` and `go-getter`: 
+ https://github.com/hashicorp/go-getter/blob/main/get.go
+ https://github.com/chartmuseum/helm-push

File: pkg/cmd/cmd_push.go [hunk 2]
```diff
@@ -127,14 +113,47 @@ func pushTarPackage(ociUrl, localTarPath string, settings *settings.Settings) er
...
 	}
 
-	// 2. Generate the OCI options from oci url and the version of current kcl package.
+	// 2. push the package
+	return pushPackage(ociUrl, kclPkg, settings)
+}
+
+// pushPackage will push the kcl package to the oci registry.
+func pushPackage(ociUrl string, kclPkg *pkg.KclPkg, settings *settings.Settings) error {
+	// 1. Package the current kcl package into default tar path.
```
Comment (Peefy) [inline pkg/cmd/cmd_push.go:122]: U can put documents to the function documents, just like:
// pushPackage will push ..
// 1.
// 2.
// 3.
// 4.

Comment (zong-zhe) [reply pkg/cmd/cmd_push.go:124]: fixed.

File: pkg/cmd/cmd_push.go (hunk 1 above)
Comment (zong-zhe) [reply pkg/cmd/cmd_push.go]: fixed.

File: pkg/cmd/cmd_push.go [hunk 3]
```diff
@@ -127,14 +111,51 @@ func pushTarPackage(ociUrl, localTarPath string, settings *settings.Settings) er
...
 	ociOpts, err := opt.ParseOciOptionFromOciUrl(ociUrl, kclPkg.GetPkgTag())
 	if err != nil {
 		return err
 	}
 
-	// 3. Push it.
-	err = oci.Push(localTarPath, ociOpts.Reg, ociOpts.Repo, ociOpts.Tag, settings)
+	reporter.Report("kpm: package '" + kclPkg.GetPkgName() + "' will be pushed.")
+	// 4. Push it.
+	err = oci.Push(tarPath, ociOpts.Reg, ociOpts.Repo, ociOpts.Tag, settings)
```
Comment (Peefy) [inline pkg/cmd/cmd_push.go:158]: After `push` successfully, there is no prompt. You can refer to `docker push` for optimization.

Comment (zong-zhe) [reply pkg/cmd/cmd_push.go:158]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**